### PR TITLE
Fix double doors' diagonal travel and inside-oriented versions

### PR DESCRIPTION
**Pull request by ShishkinP**
This one should be linked with #83

PROBLEM
1. Diagonal travel through the middle of a double door (ddoor) is disabled, because the vanilla tile engine can only handle a travel permission for a tile depending on only 1 tile (or rather the state of a structure on that tile).
2. It doesn't matter much for the sliding door/pitdoor, because structures on both of its tiles are always in the same state (open, closed or destroyed). They still disallowed it to probably make movement consistent with standard ddoors so as not to confuse the player.
3. Of the full set of ddoors only the outside-oriented half is fully-fledged. The inside-oriented only appear in Meduna tilesets with signs of being hastily copy-pasted and given up on. They behave like this:
inside-left-ddoor-bug2
The right door from the gif even has the wrong sound for closing: metal instead of plywood.
And the left-oriented are as bad and not even placed anywhere in vanilla maps.

SOLUTION
1. Fix incorrect flags for the inside ddoors at the stage of the tile db creation
2. Add new ddoor defines pointing at 2 tiles with doors instead of 1
3. Using these defines assign correct travelcost permissions at the worldbuilding stage
4. Add new logic for using them in pathfinding

TESTING

All fixed 8 doors have been tested for:
1. diagonal travel with both doors in different states (including the perceived ones)
2. destruction of adjacent walls
3. destruction of left or right part of a ddoor
4. item and room reveal through (half-)closed ddoors
5. propagation of light and smoke through (half-)closed and (half-)open ddoors
6. NPC travel (turn-based militia battles and real time)
7. attacking with a knife someone behind a closed door

Modded map (queen's palace) with all kinds of ddoors around the lobby and removed decor structures near them: https://disk.yandex.ru/d/ccPyI1d-ryPRkw
Savefile for this mod with trained militia and a soldier with TNTs: https://disk.yandex.ru/d/tIS9PGaRpMZDTg

### CODE REVIEW

File: src/game/Tactical/PathAI.cc [hunk 1]
```diff
@@ -1063,13 +1067,53 @@ INT32 FindBestPath(SOLDIERTYPE* s, INT16 sDestination, INT8 ubLevel, INT16 usMov
...
 
 					// don't let anyone path diagonally through doors!
-					if (ubCnt & 1)
-					{
-						goto NEXTDIR;
-					}
+					//if (ubCnt & 1)
+					//{
+					//	goto NEXTDIR;
+					//}
```
Comment (lynxlynxlynx) [inline src/game/Tactical/PathAI.cc:1073]: Please just delete these five lines instead.

File: src/game/Tactical/PathAI.cc [hunk 2]
```diff
@@ -2529,6 +2648,54 @@ UINT8 InternalDoorTravelCost(const SOLDIERTYPE* pSoldier, INT32 iGridNo, UINT8 u
 
 		switch( ubMovementCost )
 		{
+			case TRAVELCOST_DOORS_CLOSED_W_SW:
+				fDoorIsObstacleIfClosed = TRUE;
+				iDoorGridNo = iGridNo + DirIncrementer[WEST];
+				iDoorGridNo2 = iGridNo + DirIncrementer[SOUTHWEST];
+				ubReplacementCost = TRAVELCOST_DOOR;
+				break;
```
Comment (lynxlynxlynx) [inline src/game/Tactical/PathAI.cc:2656]: This is quite repetitive. Can the whole switch be moved to a separate function that then this and FindBestPath could use (and we get to maintain only one copy)?

File: src/game/TileEngine/TileDef.cc [hunk 3]
```diff
@@ -116,7 +117,35 @@ void CreateTileDatabase()
 			}
 
 			SetSpecificDatabaseValues(cnt1, gTileDatabaseSize, TileElement, TileSurf->bRaisedObjectType);
-
+			// fix incorrect double door flags. in vanilla it only affects DOOR3 in PALACE! tileset and DOOR1 in QUEEN'S TROPICAL
```
Comment (lynxlynxlynx) [inline src/game/TileEngine/TileDef.cc:120]: Ideally this whould be in a separate commit.

Comment (ShishkinP) [reply src/game/TileEngine/TileDef.cc:120]: Got it. I'll keep that in mind in the future.

**Comment by lynxlynxlynx** [comment]
Thanks!
